From 80e9c60c52382dc5f10d684fc3201404aaec100d Mon Sep 17 00:00:00 2001
From: William Wu <william.wu@rock-chips.com>
Date: Tue, 1 Aug 2017 15:15:00 +0800
Subject: [PATCH] usb: ehci-platform: use no relinquish port quirk only for
 rk3288

rk3288 and rk3288w use the same dts which includes no relinquish
port quirk, however rk3288w ohci can work well, so we need to add
an additional condition to disable ohci only for rk3288.

Change-Id: Ic2bd0ce577cbebe7ae2cf1b153f9e46935022f77
Signed-off-by: William Wu <william.wu@rock-chips.com>
---
 drivers/usb/host/ehci-platform.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/host/ehci-platform.c b/drivers/usb/host/ehci-platform.c
index 7866008c4350..bedf6565caa4 100644
--- a/drivers/usb/host/ehci-platform.c
+++ b/drivers/usb/host/ehci-platform.c
@@ -214,7 +214,8 @@ static int ehci_platform_probe(struct platform_device *dev)
 					  "has-transaction-translator"))
 			hcd->has_tt = 1;
 
-		if (of_property_read_bool(dev->dev.of_node,
+		if (of_machine_is_compatible("rockchip,rk3288") &&
+		    of_property_read_bool(dev->dev.of_node,
 					  "rockchip-relinquish-port"))
 			ehci_platform_hc_driver.relinquish_port =
 					  ehci_rockchip_relinquish_port;
-- 
2.35.3

